src/seedu/addressbook/data/person/Address.java

```diff
@@ -3,34 +3,52 @@
 import seedu.addressbook.data.exception.IllegalValueException;
 
 /**
- * Represents a Person's address in the address book.
- * Guarantees: immutable; is valid as declared in {@link #isValidAddress(String)}
+ * Represents a Person's address in the address book. Guarantees: immutable; is
+ * valid as declared in {@link #isValidAddress(String)} Assume the address is
+ * entered in the following format BLOCK, STREET, UNIT, POSTAL_CODE e.g.
+ * 123, Clementi Ave 3, #12-34, 231534
  */
 public class Address {
 
-    public static final String EXAMPLE = "123, some street";
+    public static final String EXAMPLE = "123, Clementi Ave 3, #12-34, 231534";
     public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses can be in any format";
     public static final String ADDRESS_VALIDATION_REGEX = ".+";
 
     public final String value;
     private boolean isPrivate;
 
+    private Block blockNum;
+    private Street streetName;
+    private Unit unitNum;
+    private PostalCode postal;
+    
     /**
      * Validates given address.
      *
-     * @throws IllegalValueException if given address string is invalid.
+     * @throws IllegalValueException
+     *             if given address string is invalid.
      */
     public Address(String address, boolean isPrivate) throws IllegalValueException {
-        String trimmedAddress = address.trim();
+        String trimmedAddress = address.trim();       
```

Comment:
There are spaces added at the back of this line.
A tip is to look through your PR once before posting it to look for these blind spots.